pkg/queue/concurrency_state.go

```diff
@@ -81,10 +81,43 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 			return
 		}
 
-		resume()
+		logger.Info("Requests increased from zero")
+		if err := resume(); err != nil {
+			logger.Errorf("Error handling resume request: %v", err)
+			panic(err)
+		}
 		paused = false
+		logger.Debug("From-Zero request successfully processed")
 		mux.Unlock()
 
 		h.ServeHTTP(w, r)
 	}
 }
+
+// ConcurrencyState Request sends to the concurrency state endpoint
+func ConcurrencyStateRequest(endpoint string, action ConcurrencyStateMessageBody) (func() error, error) {
```

Comment:
I think we'll probably want to send other info along with the freeze/pause message (pod name, for one; if for nothing else than debugging, it'd be good to know which pod is being paused and which is being resumed. Was thinking of that as a separate PR, since this is already quite large). For those reasons, I'd lean towards keeping the struct.